source/Nevermore/UpgradeScripts/Script0001-KeyAllocation.sql

```diff
@@ -27,7 +27,7 @@ BEGIN
 	if (@@ROWCOUNT = 0)
 	begin
 		INSERT INTO KeyAllocation (CollectionName, Allocated) values (@collectionName, @blockSize)
-		SELECT @blockSize
+		SELECT CAST(@blockSize as bigint) -- type must be the same as @result
```

Comment:
nit: is there any particular reason to go for casting rather than changing the type of `@blockSize` itself?

Reply:
yeah, blocksize is an input parameter to the stored procedure. Changing it to `bigint` would likely require callers in C# to change their code from `int` to `long` and be a breaking change.

We can do breaking changes if we need to, but in that case the block size doesn't need to exceed 2 billion; 32 bit int is perfectly good there, so it's less work for everyone to leave that be. This cast is the tradeoff :smile: